\documentclass[a4paper,titlepage]{book}
\usepackage[signatures, swapnames,nouppercase]{frontespizio}
\usepackage[english]{babel}
\usepackage[a4paper, top=2cm, bottom=2cm, left=2cm, right=2cm]{geometry}
\usepackage{graphicx}
\usepackage{amsmath} % or simply amstext
\usepackage{subfig}
\usepackage{graphicx}
\makeatletter
\def\@makechapterhead#1{%
  \vspace*{50\p@}%
  {\parindent \z@ \raggedright
    \normalfont
    \interlinepenalty\@M
    \Huge \bfseries\thechapter\space  #1\par\nobreak
    \vskip 40\p@
  }}
\makeatother
\newcommand{\angstrom}{\text{\normalfont\AA}}
\begin{document}
\begin{frontespizio}
\Istituzione{}
\Logo[1.5cm]{C:/Users/acer/Desktop/Tirocinio/Teoria/imm\_tn.png}
\Divisione{BACHELOR’S DEGREE IN}
\Scuola{Biomolecular Sciences and Technology}
\Titolo{Molecular Dynamics analysis to study the interaction between the human cytomegalovirus glycoprotein US9 and MICA*008}
\NCandidato{Graduant}
\Candidato{Gloria Lugoboni}
\NRelatore{Supervisor}{}
\Relatore{Prof. Emiliano Biasini}
\NCorrelatore{Tutor}{}
\Correlatore{Dr. Marta Rigoli}
\Punteggiatura{}
\Piede{Date of discussion 21/07/2022}
\end{frontespizio}

\tableofcontents

\chapter{Introduction}



\section{Metagenome Sequencing, Assembly, and Binning}
Metagenome sequencing enables the construction of metagenomes-assembled genomes (MAGs). A MAG can be seen as a microbial genome obtained by a preliminary passage of genome assembly of high quality contigs. This kind of analysis enables us to identify novel species thanks to a passage of annotation and taxonomic classification \cite{MAG}.

A typical metagenome project involves a specific pipeline, a step of sample processing and sequencing, a step of assembly and finally a step of binning followed by genome-annotation. This whole process is than completed with a statistical analysis \cite{Metagenome}.

Metagenomics is possible thanks to the study of DNA genomes, the sequencing is possible using a variety of novel sequencing technologies and platforms like Roche 454 sequencing, Illumina sequencing, and ion torrent Personal Genome Machine (PGM) \cite{Sequencing}.

Thanks to the proccess of assembly it is possible to reconstruct genomes. This method is based on a process of aligment and merging of overlapping sequences, creating large contiguous regions (contigs) \cite{Assembly}.

After the process of assembly is completed, contigs are grouped by their organism of origin into bins, using a process known as binning \cite{Binning}. The selection of high quality bins enables the identification of MAGs, these are characterized by a high completness and low levels of contamination and are used to operate taxonomic annotation and gene prediction \cite{MAG2}. These can be grouped together in the same species genome bin (SGB) if they exceed a certain threshold of nucleotide identity, with a treshold of the 5\% for genomic identity. It is possible to assign a taxonomic label based on the presence (or not) of characterized genomess \cite{SGB2}. If a genome with associated taxonomy is not availabe, we talk about known SGB (kSGB), while in the opposite case, we talk about unknown clades (uSGB) \cite{SGB}.

\section{uSGB 15132}
Which uSGB are you working with, what is your focus

We were provided with a set of 30 high-quality prebinned metagenome grouped in the same uSGB labelled SGB15132.

The bins have a completeness higher that 97.3 and the maximum redundacy registered is equal to 2.25.

DA aggiungere grafico della completeness


\chapter{Methods}
\section{Softwares and parameters used}
- What software you used for each purpose, what parameters
\subsection{Genome annotation (Prokka)}



Prokka is a fast and accurate command line software tool used to annotate prokaryoyic genomes.  It produces standards-compliant output files that can be used for further analysis or viewing in genome browsers.

Prokka expect one single input file in a FASTA format, containing an assembled genome. The process of annotation is possible thanks to the comparison of the gene codes with a large database of known sequences, identifying the best match as the most significative one and therefore associating the labelling and the relevant features to the gene codes. Prokka use this method in an hieratical manner, using initially small and reliable databases moving only at the end of the process to protein family databases.
Prokka produces several outputs file, listed in the Figure \ref{fig:prokka-output}  \cite{Prokka}.

The main input to be specified are :...

%such as the .fna file that represent a complete fasta file and contains the genome of the organism; the .faa file, that contains the amminoacids of the encoded proteins; the .tsv file that is a table summary of the annotations of the proteins with a description; finally, a .gbk file, a genome bank file.

\begin{figure}[ht]
\centering
\includegraphics[scale=0.6]{C:/Users/acer/Desktop/Appunti esami/Magistrale/Primo anno/Secondo semestre/Genomics/Microbial/Lab Lectures/Report/Table\_prokka.png}
\caption{Prokka outputs files \cite{Prokka}.}
\label{fig:prokka-output}
\end{figure}

\subsection{Pangenome analysis (Roary)}

Roary is a tool that enables the construction of large-scale prokaryote pangenomes, identifying the core and accessory genes.

The input file to Roary is a GFF file containing sequences features. 

Roary collects the coding regions from the annotated input genome. It operates a clustering process creating a network and defining a phylogenetic tree. A matrix is therefore obtained and the pangenome (core genes and accessory genes) is defined.  The process of clustering is based on the minimum percentage of identity, setted to 95\% by default. \cite{Roary}

Roary returns three graphs, the newick tree associated to the pamgenome table, a pie chart of the breakdown of genes and the number of isolate they are present in, a graph with the frequency of genes versus the number of genomes. \cite{Roary-outputs}

DA AGGIUNGERE ANALISI DEL PANGENOME SE CLOSED O OPEN
The main input to be sopecified are :...

\subsection{Taxonomic assignment (PhyloPhlAn 3.0)}

PhyloPhlAn 3.0 is an accurate and rapid tool to perform microbial genome characterization and phylogenetic analysis both of newly assembled microbial genomes and metagenomes. PhyloPhlAn 3.0 can integrate public genome resources/information to the genomes in input and is also accurate at the strain and species level. \cite{Phylo}
The main input to be sopecified are :...

\subsection{Phylogenetic analysis (Roary+FastTree)}

Roary enable us to generate a core gene alignment of our uSGB using as specific parametres -e, --mafft and -p, \textbf{roary -e --mafft -p 8 *.gff}. This aligment can be used to construct a phylogenetic tree, this is possible using FastTree, a tool for constructing large phylogenies, estimating their reliability.
FastTree exploit Neighbor-Joining and nearest neighbor interchanges to create a phylogentic tree. \cite{FastTree}

\subsection{Association with host data}

\chapter{Results and discussion}
- Description of the set of bins: where do your MAGs come from, to what SGB do they belong, completeness and contamination
\section{Genome annotation}
what functions are encoded in your MAGs?
\\Hypothetical/annotated proteins
\section{Pangenome analysis}
 what’s the size of your pangenome? Is it closed or open? How many core and accessory genes?
\section{Phylogenetic analysis and association with host data}
comparison of phylogenetic trees based on accessory gene presence/absence or on core gene alignment. Do you detect clusters of strains? How do they associate with the metadata?

\chapter{Conclusion}







\begin{thebibliography}{}
\bibitem{MAG}
Yang, Chao et al. \emph{A review of computational tools for generating metagenome-assembled genomes from metagenomic sequencing data,} Computational and structural biotechnology journal vol. 19 6301-6314. 23 Nov. 2021, \\doi:10.1016/j.csbj.2021.11.028

\bibitem{Metagenome}
Thomas T, Gilbert J, Meyer F.
\emph{Metagenomics - a guide from sampling to data analysis.} Microb Inform Exp. 2012 Feb 9;2(1):3. \\doi: 10.1186/2042-5783-2-3. PMID: 22587947; PMCID: PMC3351745.


\bibitem{Sequencing}
Reuter JA, Spacek DV, Snyder MP. \emph{High-throughput sequencing technologies.} Mol Cell. 2015 May 21;58(4):586-97.\\ doi: 10.1016/j.molcel.2015.05.004. PMID: 26000844; PMCID: PMC4494749


\bibitem{Assembly}
Churko JM, Mantalas GL, Snyder MP, Wu JC. \emph{Overview of high throughput sequencing technologies to elucidate molecular pathways in cardiovascular diseases.} Circ Res. 2013 Jun 7;112(12):1613-23. \\ doi: 10.1161/CIRCRESAHA.113.300939.

\bibitem{Binning}
Nissen, J.N., Johansen, J., Allesøe, R.L. et al. \emph{Improved metagenome binning and assembly using deep variational autoencoders.} Nat Biotechnol 39, 555–560 (2021).\\https://doi.org/10.1038/s41587-020-00777-4


\bibitem{MAG2}
Chao Yang, Debajyoti Chowdhury, Zhenmiao Zhang, William K. Cheung, Aiping Lu, Zhaoxiang Bian, Lu Zhang,
\emph{A review of computational tools for generating metagenome-assembled genomes from metagenomic sequencing data,} Computational and Structural Biotechnology Journal, Volume 19, 2021, Pages 6301-6314, ISSN 2001-0370, \\https://doi.org/10.1016/j.csbj.2021.11.028.

\bibitem{SGB2}
Pasolli E, Asnicar F, Manara S, Zolfo M, Karcher N, Armanini F, Beghini F, Manghi P, Tett A, Ghensi P, Collado MC, Rice BL, DuLong C, Morgan XC, Golden CD, Quince C, Huttenhower C, Segata N. \emph{Extensive Unexplored Human Microbiome Diversity Revealed by Over 150,000 Genomes from Metagenomes Spanning Age, Geography, and Lifestyle.} Cell. 2019 Jan 24;176(3):649-662.e20. \\doi: 10.1016/j.cell.2019.01.001.

\bibitem{SGB}
Blanco-Míguez, A., Beghini, F., Cumbo, F. et al. \emph{Extending and improving metagenomic taxonomic profiling with uncharacterized species using MetaPhlAn 4.} Nat Biotechnol (2023). \\https://doi.org/10.1038/s41587-023-01688-w

\bibitem{Prokka}
Torsten Seemann,\emph{Prokka: rapid prokaryotic genome annotation,} Bioinformatics, Volume 30, Issue 14, July 2014, Pages 2068–2069,\\ https://doi.org/10.1093/bioinformatics/btu153

\bibitem{Roary}
Page AJ, Cummins CA, Hunt M, Wong VK, Reuter S, Holden MT, Fookes M, Falush D, Keane JA, Parkhill J. \emph{Roary: rapid large-scale prokaryote pan genome analysis.} Bioinformatics. 2015 Nov 15; 31(22):3691-3. \\doi: 10.1093/bioinformatics/btv421

\bibitem{Roary-outputs}
\emph{https://sanger-pathogens.github.io/Roary/}

\bibitem{Phylo}
Asnicar, F., Thomas, A.M., Beghini, F. et al.
\emph{Precise phylogenetic analysis of microbial isolates and genomes from metagenomes using PhyloPhlAn 3.0.} Nat Commun 11, 2500 (2020).\\ https://doi.org/10.1038/s41467-020-16366-7

\bibitem{FastTree}
Morgan N. Price, Paramvir S. Dehal, Adam P. Arkin, \emph{FastTree: Computing Large Minimum Evolution Trees with Profiles instead of a Distance Matrix, Molecular Biology and Evolution,} Volume 26, Issue 7, July 2009, Pages 1641–1650, \\https://doi.org/10.1093/molbev/msp077


\end{thebibliography}




\end{document}