\documentclass[a4paper,titlepage]{book}
\usepackage[signatures, swapnames,nouppercase]{frontespizio}
\usepackage[english]{babel}
\usepackage[a4paper, top=2cm, bottom=2cm, left=1.5cm, right=1.5cm]{geometry}
\usepackage{graphicx}
\usepackage{amsmath} % or simply amstext
\usepackage{subfig}
\usepackage{graphicx}
\usepackage{xcolor}
\definecolor{light-gray}{gray}{0.95}
\newcommand{\code}[1]{\colorbox{light-gray}{\texttt{#1}}}
\makeatletter
\def\@makechapterhead#1{%
  \vspace*{50\p@}%
  {\parindent \z@ \raggedright
    \normalfont
    \interlinepenalty\@M
    \Huge \bfseries\thechapter\space  #1\par\nobreak
    \vskip 40\p@
  }}
\makeatother
\newcommand{\angstrom}{\text{\normalfont\AA}}
\begin{document}
\begin{frontespizio}
\Istituzione{}
\Logo[1.5cm]{C:/Users/acer/Desktop/Tirocinio/Teoria/imm\_tn.png}
\Divisione{BACHELORâ€™S DEGREE IN}
\Scuola{Biomolecular Sciences and Technology}
\Titolo{Molecular Dynamics analysis to study the interaction between the human cytomegalovirus glycoprotein US9 and MICA*008}
\NCandidato{Graduant}
\Candidato{Gloria Lugoboni}
\NRelatore{Supervisor}{}
\Relatore{Prof. Emiliano Biasini}
\NCorrelatore{Tutor}{}
\Correlatore{Dr. Marta Rigoli}
\Punteggiatura{}
\Piede{Date of discussion 21/07/2022}
\end{frontespizio}

\tableofcontents

\chapter{Introduction}



\section{Metagenome Sequencing, Assembly, and Binning}
Metagenome sequencing enables the construction of metagenomes-assembled genomes (MAGs). A MAG can be seen as a microbial genome obtained by a preliminary passage of genome assembly of high quality contigs. This kind of analysis enables us to identify novel species thanks to a passage of annotation and taxonomic classification \cite{MAG}.

A typical metagenome project involves a specific pipeline, a step of sample processing and sequencing, a step of assembly and finally a step of binning followed by genome-annotation. This whole process is than completed with a statistical analysis \cite{Metagenome}.

Metagenomics is possible thanks to the study of DNA genomes, the sequencing is possible using a variety of novel sequencing technologies and platforms like Roche 454 sequencing, Illumina sequencing, and ion torrent Personal Genome Machine (PGM) \cite{Sequencing}.

Thanks to the proccess of assembly it is possible to reconstruct genomes. This method is based on a process of aligment and merging of overlapping sequences, creating large contiguous regions (contigs) \cite{Assembly}.

After the process of assembly is completed, contigs are grouped by their organism of origin into bins, using a process known as binning \cite{Binning}. The selection of high quality bins enables the identification of MAGs, these are characterized by a high completness and low levels of contamination and are used to operate taxonomic annotation and gene prediction \cite{MAG2}. These can be grouped together in the same species genome bin (SGB) if they exceed a certain threshold of nucleotide identity, with a treshold of the 5\% for genomic identity. It is possible to assign a taxonomic label based on the presence (or not) of characterized genomess \cite{SGB2}. If a genome with associated taxonomy is not availabe, we talk about known SGB (kSGB), while in the opposite case, we talk about unknown clades (uSGB) \cite{SGB}.



With the term pangenome, we indicate the union of the \textit{core genome}, containing genes present in all strains, and the \textit{dispensable genome}, also called \textit{accessory genome}, containing genes present in two or more strains and genes unique to single strains \cite{Medini}.


DA SPIEGARE IL PANGENOME ANAYSIS E LE ALTRE ANALISI CHE FACCIAMO??



\chapter{Methods}
\section{Softwares and parameters used}
\subsection{Genome annotation (Prokka)}

Prokka is a fast and accurate command line software tool used to annotate prokaryoyic genomes.  It produces standards-compliant output files that can be used for further analysis or viewing in genome browsers.

Prokka expect one single input file in a FASTA format, containing an assembled genome. The process of annotation is possible thanks to the comparison of the gene codes with a large database of known sequences, identifying the best match as the most significative one and therefore associating the labelling and the relevant features to the gene codes. Prokka use this method in an hieratical manner, using initially small and reliable databases moving only at the end of the process to protein family databases.
Prokka produces several outputs file, listed in the Figure \ref{fig:prokka-output}  \cite{Prokka}.

We need to specify several parameters, specifically, the input files, our MAGs, the output directory \code{--outdir} and the parameter \code{--kingdom Bacteria} that is needed to specify the annotation mode, to make the prokka more fast.
\\ \newline \\
\code{prokka --kingdom Bacteria --outdir SGB15132\_prokka\_output .f*}.

%such as the .fna file that represent a complete fasta file and contains the genome of the organism; the .faa file, that contains the amminoacids of the encoded proteins; the .tsv file that is a table summary of the annotations of the proteins with a description; finally, a .gbk file, a genome bank file.

\begin{figure}[ht]
\centering
\includegraphics[scale=0.6]{Table\_prokka.png}
\caption{Prokka outputs files \cite{Prokka}.}
\label{fig:prokka-output}
\end{figure}

\subsection{Pangenome analysis and Phylogenetic analysis (Roary, Roary + FastTree)}

Roary is a tool that enables the construction of large-scale prokaryote pangenomes, identifying the core and accessory genes.

The input file to Roary is a GFF file containing sequences features. 

Roary collects the coding regions from the annotated input genome. It operates a clustering process creating a network and defining a phylogenetic tree. A matrix is therefore obtained and the pangenome (core genes and accessory genes) is defined.  The process of clustering is based on the minimum percentage of identity, setted to 95\% by default \cite{Roary}.

The main output of Roary is a tree obtained using the presence and absence of the accessory genes. It is a tree used to have an initial insight of the data, grouping in a quick way the genomes based on their accessory genes \cite{Roary}. It can be visualized using iTOL, an online tool for phylogenetic tree display \cite{iTOL}.

Roary returns three graphs, the newick tree associated to the pangenome table, a pie chart of the breakdown of genes and the number of isolate they are present in, a graph with the frequency of genes versus the number of genomes. \cite{Roary-outputs}


There are some main parameters that need to be specified to roary, specifically, the input \code{.gff} files; the output directory \code{-f roary\_out}; the , \code{-i} parameter, specifing the percentage identity of blastp, here used at 95\%; the , \code{-cd} parameter, percentage of isolates a gene must be in to be considered part of the core genome, here setted at 90\%.
\\ \newline \\ \code{roary .gff -i 95 -cd 90} \\ \newline \\


With Roary it is also possible to perform a core gene alignment to generate a more reliable tree. A core-genome alignment is more scalable with respect to the whole-genome alignment. It is an alignment useful to identify the core genes conserved in all aligned genomes and that can be very useful to infer the phylogeny \cite{Core-align}.

The main parameters sto be specified are the \code{-e} parameter,needed to perform a core gene alignment; the \code{-n} parameter, to use mafft as the tool for the multiple sequence alignmet, making the process faster and finally the parameter \code{-p}, needed to specify the number of threads, increasing therefore the speed \cite{Roary}.
\\
\newline
\\
\code{roary .gff -i 95 -cd 90 -e -n -p 8}.
\\
\newline
\\
The core gene aligment can be used to construct a phylogenetic tree. This is possible using FastTree, a tool for constructing large phylogenies, estimating their reliability. FastTree exploit Neighbor-Joining and nearest neighbor interchanges to create a phylogentic tree. \cite{FastTree}
Specifically, we used FastTreeMP, that allow the parallelization of the steps needed in computing a tree \cite{FTMP}.
The tree is obtained using the following code,\\ \newline \\ \code{FastTreeMP -gtr -nt -out core\_gene.tre core\_gene\_alignment.aln},\\ \newline \\ in which the parameter \code{-gtr} express the generalized time-reversible model (to be used with nucleotide alignments only) while the parameter \code{-nt} is used to specify that the alignment is performed on nucleotides.




\subsection{Taxonomic assignment (PhyloPhlAn 3.0)}

PhyloPhlAn 3.0 is an accurate and rapid tool to perform microbial genome characterization and phylogenetic analysis both of newly assembled microbial genomes and metagenomes. PhyloPhlAn 3.0 can integrate public genome resources/information to the genomes in input and is also accurate at the strain and species level and allow the assign to each bin obtained via metagenomic assembly its closest species-level genome bins \cite{Phylo}.
There are some main parameters needed to be specified, mainly, the input folder with the \code{-i} parameter; the output folder, with the parameter \code{-o}; the \code{--nproc} parameter, used to specify the CPUs that can be used; the \code{-n} parameter that allow us to decide  how many SGBs (sorted by increasing average genomic distance) will be reported for each input bin in the output file; the \code{--database\_update} parameter to update the databases file, the \code{-d} parameter to specify the name of the output database adn finally, the \code{--verbose} parameter to print to the bash \cite{PhyloGuide}.
The final command is the following:\\ \newline \\ \code{phylophlan\_metagenomic -i phylo -o phylo\_out --nproc 4 -n 1 --database\_update -d CMG2324 --verbose}

\chapter{Results and discussion}

\section{uSGB 15132}

We were provided with a set of 30 high-quality prebinned metagenomes grouped in the same uSGB labelled SGB15132.

As shown in the Figure \ref{fig:compl} the bins have a completeness higher that 97.3 and the maximum redundacy registered is equal to 2.25.
The avarage values are : ..... as can be seen in the Figure \ref{fig:avarage}.

\begin{figure}[ht]
\centering
\includegraphics[scale=0.7]{completeness\_distribution.png}
\caption{Completeness distribution of the given MAGs.}
\label{fig:compl}
\end{figure}

\section{Genome annotation}
After the gene annotation process, possible using prokka, we were able to identify that the number of the CDS (protein coding sequence) is slightly variable, spanning from a minimun value of 2651 to a max of 3935. For each MAG, as shown in the Figure \ref{fig:CDS} more or less a half of the CDS are known proteins, while the other half is represented by RNA or hypotetical proteins.


\begin{figure}[ht]
\centering
\includegraphics[scale=0.4]{CDS\_hypothetical\_knowns.png}
\caption{Main annotation results}
\label{fig:CDS}
\end{figure}

\section{Pangenome analysis}



Each SGB strain was found to contain an average of 1317 genes that are present in every strain (core genome), plus 8407 genes that are absent in more than one strains (strain $<$ 30) (accessory genome). This can be seen in the Figure \ref{fig:pangenome2}, showing the frequency plot of the genes per genome, this plot gives a general overview of the frequency of genes within a whole genome set, typically these plots have an shape and most genes can be detected in a single genome or in all genomes \cite{Analysis-roary}. Here it is shown that the number of genes present in all the genomes correspond to the number of core genes.
The accessory genes are also divided into genes present in only one strain (cloud genome) or genes present in two or more strains but not all strains (shell genome) \cite{Medini}. The figure \ref{fig:pangenome1} show the subparts constituting the pangenome, associated with the total number of genes.



\begin{figure}[ht]
\centering
\includegraphics[scale=0.6]{Graphs/pangenome\_frequency.png}
\caption{Pangenome Frequency}
\label{fig:pangenome2}
\end{figure}

\begin{figure}[ht]
\centering
\includegraphics[scale=0.5]{Graphs/pangenome\_pie.png}
\caption{Core and accessory genome}
\label{fig:pangenome1}
\end{figure}


Looking at the Figures \ref{fig:pangenome3} and  \ref{fig:pangenome4} we can obtain/derive an idea on the pangenome, specifically if it is open or closed. It is important to remember that these results were obtained using only 30 metagenomes, making the process of discussion more complex. Even though, looking at the Conserved vs Total genes plot in Figure \ref{fig:pangenome3}, we can argue that the total genes initial slope is very high but at the end, when almost all the genomes were added, the slope start to decrease. The conserved gene line is almost stationary, with litte changes when the genomes are all added. Only looking at this graph it is difficult to define if the pangenome is open or closed, therefore, the observation of the Unique vs New genes plot can be useful. Looking at the Figure \ref{fig:pangenome4}, we can see that, adding new genomes doesn't give new information, the number of new genes is almost at zero when we add the last genome. This help us to state with a good security that the pangenome is closed.


\begin{figure}[ht]
\centering
\includegraphics[scale=0.4]{Graphs/conserved\_vs\_total\_genes.png}
\caption{Conserved vs Total genes}
\label{fig:pangenome3}
\end{figure}

\begin{figure}[ht]
\centering
\includegraphics[scale=0.4]{Graphs/unique\_vs\_new\_genes.png}
\caption{Unique vs New genes}
\label{fig:pangenome4}
\end{figure}

\section{Phylogenetic analysis and association with host data}
PhyloPhlAn enable us to operate a phylogenetic analysis. The main output correspond to a list of the closest SGBs sorted by their increasing average sequence distance (Mash distance) in a tab-separated file \cite{PhyloGuide}. The information of each SGB contains specific columns:
\\
\\ \code{my\_bin}: is the input bin name
\\ \code{(k|u)SGB\_ID}: the SGB ID, \textit{k} indicate a known SGB, \textit{u} indicate an unknown SGB
\\ \code{taxa\_level}: taxonomic level the SGB has been assigned to
\\ \code{taxonomy}: the full taxonomic label assigned to the SGB
\\ \code{average\_mash\_distance}: the distance is calculated with respect to all the genomes in the SGB.

In our case, for each MAG we get the same taxa\_level, the species Flavonifractor plautii, a strictly anaerobic rod shaped bacterium. It is a commensal of the human intestinal microbiota which is  \cite{Plauti}


\chapter{Conclusion}
Thanks to this project it was possible to obtain an insight on the metagenome workflow and the main analysis that are neeed when dealing with unkown genomes.
We initially investigated the uSGB we were given computing general statistics on the quality and the associated metadata.
We then performed a process of gene annotation and retrived informations about the number of coding sequences and hypotetical proteins in these CDS (Figure \ref{fig:CDS}).
We than were able to perform a pangenome analysis and hypotize that the pangenome of our uSGB is closed as the addition of new genomes gives us little to no information as shown in the Figure \ref{fig:pangenome4}. Also, it was possible to identify the genes shared by at least ??TO CHECK?? 90\% of our MAGs, the core genome. These correspons to $\sim$ 15\% of the total gene.
From the taxonomix assigment of the uSGB it was possible to identify the \textit{Oscillospiraceae} family and specifically the \textit{Flavonifractor plautii} species.


Periprosthetic Hip Joint Infection with Flavonifractor plautii: A Literature Review and Case Report 
NB Artite remautoide Ã¨ un tipo di Joint infection!!

Flavonifractor, the sole member of Ruminococcaceae that was associated with the depressed subjects, has previously been linked to lower quality of life scores and MDD18,23 as well as bipolar disorder15 and generalized anxiety disorder12. This genus, which currently includes the single named species F. plautii (formerly Clostridium orbiscindens and Eubacterium plautii)185, has also previously been linked to various autoimmune disorders3,7,186,187, chronic kidney disease188, and colorectal cancer189,190. Furthermore, F. plautii was demonstrated in vitro to have epithelial invasive potential191. Therefore, there is significant evidence to suggest that unlike its generally anti-inflammatory relatives, Flavonifractor may be associated with disease despite its ability to produce butyrate under some conditions185,192. As suggested in other work15,190, this is possibly related to the genusâ€™ eponymous capacity for cleaving flavonoids that reach the colon, including antioxidants such as quercetin, although disentangling the impacts of flavonoids and their microbial breakdown products on inflammation in vivo is difficult 
DA ARTICOLO: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7415740/







\begin{thebibliography}{}
\bibitem{MAG}
Yang, Chao et al. \emph{A review of computational tools for generating metagenome-assembled genomes from metagenomic sequencing data,} Computational and structural biotechnology journal vol. 19 6301-6314. 23 Nov. 2021, \\doi:10.1016/j.csbj.2021.11.028

\bibitem{Metagenome}
Thomas T, Gilbert J, Meyer F.
\emph{Metagenomics - a guide from sampling to data analysis.} Microb Inform Exp. 2012 Feb 9;2(1):3. \\doi: 10.1186/2042-5783-2-3. PMID: 22587947; PMCID: PMC3351745.


\bibitem{Sequencing}
Reuter JA, Spacek DV, Snyder MP. \emph{High-throughput sequencing technologies.} Mol Cell. 2015 May 21;58(4):586-97.\\ doi: 10.1016/j.molcel.2015.05.004. PMID: 26000844; PMCID: PMC4494749


\bibitem{Assembly}
Churko JM, Mantalas GL, Snyder MP, Wu JC. \emph{Overview of high throughput sequencing technologies to elucidate molecular pathways in cardiovascular diseases.} Circ Res. 2013 Jun 7;112(12):1613-23. \\ doi: 10.1161/CIRCRESAHA.113.300939.

\bibitem{Binning}
Nissen, J.N., Johansen, J., AllesÃ¸e, R.L. et al. \emph{Improved metagenome binning and assembly using deep variational autoencoders.} Nat Biotechnol 39, 555â€“560 (2021).\\https://doi.org/10.1038/s41587-020-00777-4


\bibitem{MAG2}
Chao Yang, Debajyoti Chowdhury, Zhenmiao Zhang, William K. Cheung, Aiping Lu, Zhaoxiang Bian, Lu Zhang,
\emph{A review of computational tools for generating metagenome-assembled genomes from metagenomic sequencing data,} Computational and Structural Biotechnology Journal, Volume 19, 2021, Pages 6301-6314, ISSN 2001-0370, \\https://doi.org/10.1016/j.csbj.2021.11.028.

\bibitem{SGB2}
Pasolli E, Asnicar F, Manara S, Zolfo M, Karcher N, Armanini F, Beghini F, Manghi P, Tett A, Ghensi P, Collado MC, Rice BL, DuLong C, Morgan XC, Golden CD, Quince C, Huttenhower C, Segata N. \emph{Extensive Unexplored Human Microbiome Diversity Revealed by Over 150,000 Genomes from Metagenomes Spanning Age, Geography, and Lifestyle.} Cell. 2019 Jan 24;176(3):649-662.e20. \\doi: 10.1016/j.cell.2019.01.001.

\bibitem{SGB}
Blanco-MÃ­guez, A., Beghini, F., Cumbo, F. et al. \emph{Extending and improving metagenomic taxonomic profiling with uncharacterized species using MetaPhlAn 4.} Nat Biotechnol (2023). \\https://doi.org/10.1038/s41587-023-01688-w


\bibitem{Medini}
Medini, Duccio et al. \emph{The microbial pan-genome.} Current opinion in genetics and development vol. 15,6 (2005): 589-94. \\doi:10.1016/j.gde.2005.09.006

\bibitem{Prokka}
Torsten Seemann,\emph{Prokka: rapid prokaryotic genome annotation,} Bioinformatics, Volume 30, Issue 14, July 2014, Pages 2068â€“2069,\\ https://doi.org/10.1093/bioinformatics/btu153

\bibitem{Roary}
Page AJ, Cummins CA, Hunt M, Wong VK, Reuter S, Holden MT, Fookes M, Falush D, Keane JA, Parkhill J. \emph{Roary: rapid large-scale prokaryote pan genome analysis.} Bioinformatics. 2015 Nov 15; 31(22):3691-3. \\doi: 10.1093/bioinformatics/btv421

\bibitem{iTOL}
Ivica Letunic, Peer Bork, \emph{Interactive Tree Of Life (iTOL) v5: an online tool for phylogenetic tree display and annotation}, Nucleic Acids Research, Volume 49, Issue W1, 2 July 2021, Pages W293â€“W296, \\https://doi.org/10.1093/nar/gkab301

\bibitem{Roary-outputs}
\emph{https://sanger-pathogens.github.io/Roary/}

\bibitem{Core-align}
Treangen, T.J., Ondov, B.D., Koren, S. et al. \emph{The Harvest suite for rapid core-genome alignment and visualization of thousands of intraspecific microbial genomes.} Genome Biol 15, 524 (2014).\\https://doi.org/10.1186/s13059-014-0524-x

\bibitem{FastTree}
Morgan N. Price, Paramvir S. Dehal, Adam P. Arkin, \emph{FastTree: Computing Large Minimum Evolution Trees with Profiles instead of a Distance Matrix, Molecular Biology and Evolution,} Volume 26, Issue 7, July 2009, Pages 1641â€“1650, \\https://doi.org/10.1093/molbev/msp077


\bibitem{FTMP}
Price MN, Dehal PS, Arkin AP, \emph{FastTree 2â€“approximately maximum-likelihood trees for large alignments}. PLoS One. 2010, 5\\doi: e9490-10.1371/journal.pone.0009490.


\bibitem{Phylo}
Asnicar, F., Thomas, A.M., Beghini, F. et al.
\emph{Precise phylogenetic analysis of microbial isolates and genomes from metagenomes using PhyloPhlAn 3.0.} Nat Commun 11, 2500 (2020).\\ https://doi.org/10.1038/s41467-020-16366-7

\bibitem{PhyloGuide}
\emph{https://github.com/biobakery/phylophlan/wiki}


\bibitem{Analysis-roary}
\emph{https://help.ezbiocloud.net/gene-frequency-plot-in-pan-genome/}

\bibitem{Plauti}
Berger FK, Schwab N, Glanemann M, Bohle RM, GÃ¤rtner B, Groesdonk HV. \emph{Flavonifractor (Eubacterium) plautii bloodstream infection following acute cholecystitis.} IDCases. 2018 Oct 28;14:e00461. \\doi: 10.1016/j.idcr.2018.e00461.


\end{thebibliography}




\end{document}